qualification/antenna_channelised_voltage/test_gain.py

```diff
@@ -107,3 +107,57 @@ async def next_chunk_data() -> NDArray[np.complex128]:
     # quantisation, since gain is handled as single-precision float.
     with check:
         assert max_rel_error < 0.1, "Maximum error exceeds 0.5 dB"
+
+
+@pytest.mark.name("Ordering of gains and capture-start")
+@pytest.mark.no_capture_start("baseline-correlation-products")
+async def test_gains_capture_start(
+    cbf: CBFRemoteControl,
+    receive_baseline_correlation_products: BaselineCorrelationProductsReceiver,
+    pdf_report: Reporter,
+) -> None:
+    r"""Test that gains applied before capture-start are not delayed.
+
+    Verification methods
+    --------------------
+    Verified by test. Change the gains, then immediately issue a capture-start
+    request. Verify that the received data reflects the change in gains.
+    """
+    receiver = receive_baseline_correlation_products
+    pcc = cbf.product_controller_client
+
+    pdf_report.step("Inject white noise on first antenna.")
+    signals = "common = wgn(0.1); common; common;"
+    await cbf.dsim_clients[0].request("signals", signals)
+    pdf_report.detail(f"Set dsim signals to {signals}.")
+
+    pdf_report.step("Wait for injected signal to reach F-engine.")
+    label = receiver.input_labels[0]
+    for _ in range(10):
+        power = await pcc.sensor_value(f"antenna-channelised-voltage.{label}.dig-rms-dbfs", float)
+        pdf_report.detail(f"dig-rms-dbfs = {power:.3g} dBFS")
+        if power > -1000:
+            break
```

Comment:
`-1000`, arbitrarily selected?

Reply:
Honestly, can't think of anything "less arbitrary", unfortunately. I'd say an explainer comment should be sufficient (like you've outlined in your reply). I don't have a sense of 'expected values of interest', but this does seem like a decent threshold.

Reply:
It occurs to me that a better approach to detecting this is just to use the F-engine's `<input>.rx.timestamp` sensor. If I was ambitious I could even use that to estimate how long to sleep for. I'll update when I've done that.

Reply:
> If I was ambitious I could even use that to estimate how long to sleep for.

The catch is that the sensor is only updated once per second, so this would give a very inaccurate estimate. I've stuck with the existing simple approach of just sleeping 0.5s and checking again.